source/speech/speech.py

```diff
@@ -1169,9 +1171,34 @@ def speak(  # noqa: C901
 			)
 			if not inCharacterMode:
 				speechSequence[index] += CHUNK_SEPARATOR
+				if (
+					config.conf["speech"]["reportLanguage"]
+					and curLanguage not in (defaultLanguage, _speechState.lastReportedLanguage)
+					and languageHandler.getLanguageDescription(curLanguage) not in speechSequence
+					and languageHandler.getLanguageDescription(curLanguage) is not None
```

Comment:
What happens if language descriptions is not known by Windows? Could we report the language code instead?

Reply:
> @nvdaes - can you please answer this question from @CyrilleB79 [#17685 (comment)](https://github.com/nvaccess/nvda/pull/17685#discussion_r1989581692)


Yes Cyrille. We have in `source/speech/languageHandling.py`
```
		langDesc = languageHandler.getLanguageDescription(item.lang)
		if langDesc is None:
			langDesc = item.lang
```